core/src/auctions.ts

```diff
@@ -221,6 +224,15 @@ export const enrichAuctionWithPriceDropAndMarketDataRecords = async function (
     return await enrichAuctionWithMarketDataRecords(auctionWithFees, network, useAutoRouter);
 };
 
+export const enrichAuctionWithPriceDropAndMarketDataRecords = memoizee(
+    _enrichAuctionWithPriceDropAndMarketDataRecords,
+    {
+        maxAge: MARKET_DATA_CACHE_TIME,
+        promise: true,
+        length: 3,
+    }
+);
```

Comment:
Why `_getMarketPrice` doesn't work? Why did you have to add another layer of caching here? How does caching of the `auction` work here? By the object values (serialised) or by the object reference (so no matter if the values change or not)?

Reply:
> Why _getMarketPrice doesn't work? 

essentially it narrows down to having `getMarketDataRecords` in the calltree of `enrichAuctionWithPriceDropAndMarketDataRecords` which is responsible for firing a lot of requests to update the info.

> How does caching of the auction work here? By the object values (serialised) or by the object reference (so no matter if the values change or not)?

i sense a rhetorical question that is designed to send me on a journey of past memories. I recall a claim about object reference being memorized and changes to the object not being tracked.

with both points above it seems like caching `getMarketDataRecords` instead would be a better solution that the current one.